### PR TITLE
jmhCompile should not compile with compiler arg -Werror

### PR DESCRIPTION
**Pull request by DotSpy**
Jmh fail on compile time because of -Werror flag

### CODE REVIEW

**Comment by jkwatson** [comment]
we should add `jmhCompile` to the CI steps, so this doesn't happen again. Can you add this to the circle configuration, or to the Makefile, @DotSpy ?

**Comment by DotSpy** [comment]
i would like to get into CI, i'll do it

**Comment by DotSpy** [comment]
@jkwatson done

File: Makefile [hunk 1]
```diff
@@ -3,6 +3,7 @@
 .PHONY: test
 test:
 	./gradlew clean assemble check --stacktrace
+	./gradlew compileJmhJava
```
Comment (thisthat) [inline Makefile]: I would create a different Makefile target for the benchmarks. This way, we can decouple benchmarks from compilation in CircleCI

Comment (DotSpy) [reply Makefile]: agree

**Comment by DotSpy** [comment]
Any ideas why on CI 
https://circleci.com/gh/open-telemetry/opentelemetry-java/2008?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
> Task :opentelemetry-api:compileJmhJava SKIPPED ?

File: Makefile [hunk 2]
```diff
@@ -4,6 +4,10 @@
 test:
 	./gradlew clean assemble check --stacktrace
 
+.PHONY: becnhmark
+becnhmark:
```
Comment (thisthat) [inline Makefile]: nit: typo

File: .circleci/config.yml [hunk 3]
```diff
@@ -6,6 +6,9 @@ init_task: &init_task
 build_task: &build_task
   name: Build
   command: make test
+compile_benchmark_task: &compile_benchmark_task
+  name: Compile JMH
+  command: make becnhmark
```
Comment (thisthat) [inline .circleci/config.yml]: nit: typo

**Comment by thisthat** [comment]
> Any ideas why on CI
> [circleci.com/gh/open-telemetry/opentelemetry-java/2008?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link](https://circleci.com/gh/open-telemetry/opentelemetry-java/2008?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)
> 
> > Task :opentelemetry-api:compileJmhJava SKIPPED ?

In the link you provide, you can find the `compileJmhJava` task be executed after the build task.

**Comment by DotSpy** [comment]
> 
> 
> > Any ideas why on CI
> > [circleci.com/gh/open-telemetry/opentelemetry-java/2008?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link](https://circleci.com/gh/open-telemetry/opentelemetry-java/2008?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)
> > > Task :opentelemetry-api:compileJmhJava SKIPPED ?
> 
> In the link you provide, you can find the `compileJmhJava` task be executed after the build task.

yeap it executed in Build section, but i think that it should be executed in Compile JMH section and not executed in Build

**Comment by thisthat** [comment]
Sorry, I wrongly read your comment. It is skipped because there is no `/src/jmh/java` directory in the API module.

**Comment by DotSpy** [comment]
still if u look at build section it calling > Task :opentelemetry-sdk:compileJmhJava

**Comment by DotSpy** [comment]
Found: gradle assemble calling :opentelemetry-sdk:compileJmhJava

**Comment by DotSpy** [comment]
@jkwatson as u can see here https://circleci.com/gh/open-telemetry/opentelemetry-java/2008#tests/containers/0
gradle assemble calling > Task :opentelemetry-sdk:compileJmhJava
so now we have jmhCompile in our CI, and after i will update https://github.com/open-telemetry/opentelemetry-java/pull/815 from master it will fail.
So question is : Should we move out jmhCompile from gradle assemble and build step in CI in this PR or should we not touch it for now?

**Comment by jkwatson** [comment]
This seems fine to me.

**Comment by DotSpy** [comment]
in current state there is no sense for Compile JMH step, in it we will get
> Task :opentelemetry-sdk:compileJmhJava UP-TO-DATE
because it's called at Build step

**Pull request merged**